python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1009,15 +991,24 @@ def _test_write_to_dataset_no_partitions(base_path,
     output_table = pa.Table.from_pandas(output_df)
 
     if filesystem is None:
-        filesystem = LocalFileSystem._get_instance()
+        filesystem = LocalFileSystem()
 
     # Without partitions, append files to root_path
     n = 5
     for i in range(n):
         pq.write_to_dataset(output_table, base_path,
                             filesystem=filesystem)
-    output_files = [file for file in filesystem.ls(str(base_path))
-                    if file.endswith(".parquet")]
+
+    try:
+        output_files = [file for file in filesystem.ls(str(base_path))
```

Comment:
Does this ever succeed on non-legacy filesystems?

Reply:
No, `.ls()` method will always fail. WIll remove the try case 👍

Reply:
Ah sorry, in the case of 'S3FileSystem' the try is needed. Or I can do the same as in `_generate_partition_directories` and wrap it into pyarrow Filesystem in this case?

Reply:
https://github.com/apache/arrow/pull/39825/commits/228391ef05dc4a6ea61f689e9fdbc759f4097711